src/main/java/frc5124/robot2020/subsystems/Shooter.java

```diff
@@ -2,160 +2,73 @@
-        if((max != kMaxOutput) || (min != kMinOutput)) { 
-            m_pidController.setOutputRange(min, max); 
-            kMinOutput = min; 
-            kMaxOutput = max; 
-        }
-
-        /**
-         * PIDController objects are commanded to a set point using the 
-         * SetReference() method.
-         * 
-         * The first parameter is the value of the set point, whose units vary
-         * depending on the control type set in the second parameter.
-         * 
-         * The second parameter is the control type can be set to one of four 
-         * parameters:
-         *  com.revrobotics.ControlType.kDutyCycle
-         *  com.revrobotics.ControlType.kPosition
-         *  com.revrobotics.ControlType.kVelocity
-         *  com.revrobotics.ControlType.kVoltage
-         */
-
-        m_pidController.setReference(setPoint, ControlType.kVelocity);
+/**
+ * 
+ * WARNING 
+ * Untuned
+ * 
+ * 
+ * 
+ */
+public class Shooter implements Subsystem {
+  private CANSparkMax shootMotorFollower = new CANSparkMax(RobotMap.ShooterMap.shootFollowerCanID, MotorType.kBrushless);
+  private CANSparkMax shootMotorLeader = new CANSparkMax(RobotMap.ShooterMap.shootLeaderCanID, MotorType.kBrushless);
+  private CANPIDController shootController; 
+  private double currentVelocity;
+ 
 
-        SmartDashboard.putNumber("ProcessVariable", m_encoder.getVelocity());
+  
+  public Shooter() {
+    shootMotorFollower.follow(shootMotorLeader);
+    shootController = shootMotorLeader.getPIDController();
+    shootController.setD(RobotMap.ShooterMap.Kd);
+    shootController.setFF(RobotMap.ShooterMap.Kf);
+    shootController.setP(RobotMap.ShooterMap.Kp);
+    shootController.setI(RobotMap.ShooterMap.Ki);
+    shootController.setOutputRange(-RobotMap.ShooterMap.maxVelocity, RobotMap.ShooterMap.maxVelocity);
+  }
+
+  /**
+   * @param targetVelocity desired ft/s [advise 30 or 0]
+   */
+  public void setTargetVelocity(double targetVelocity) {
+    targetVelocity = (targetVelocity / RobotMap.ShooterMap.conversionConstant);
+    if (shootMotorLeader.getAppliedOutput() == 0 && targetVelocity == 0) {
+      return;
     }
+      shootController.setReference(targetVelocity, ControlType.kVelocity);
+  }
+
+/**
+ * Units of ft/s
+ */
+  public double getVelocity() {
+    return ((shootMotorLeader.getEncoder().getVelocity())  * RobotMap.ShooterMap.conversionConstant); 
+   }
+  
+  /**
+   * @deprecated
+   */
+  public void directPower (double power) {
+    shootMotorLeader.set(power);
+  }
+
+  
+  @Override
+  public void periodic() {
+  }
 
-    // Put methods for controlling this subsystem
-    // here. Call these from Commands.
- 
-     public void spinUp() {
-        setPoint = 0.6 * maxRPM;
-        m_pidController.setReference(setPoint, ControlType.kVelocity);
-     }
- 
-     public void spinDown() {
-        setPoint = 0.0;
-        m_pidController.setReference(setPoint, ControlType.kVelocity);
-     }
-     
-     public boolean atSpeed() {
```

Comment:
I think we need a boolean method that a command might call be told if the shooter is at shooting speed or not

Reply:
alright, will add